### PR TITLE
Update the Zaraz documentation

### PR DESCRIPTION
**Pull request by bjesus**
This adds documentation for new Zaraz functionalities, such as import & export and Worker Variables. It also adds some new FAQ entries.

### CODE REVIEW

File: content/zaraz/advanced/worker-variables.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+2. A Worker Variable that takes a cookie, makes a request to your backend, and returns the User ID
+3. A Worker Variable that hashes a value before sending it to a third-party vendor
+
+{{<Aside type="note">}}
+Worker Variables are only available for accounts on a [Workers Paid plan](/workers/platform/pricing/).
+{{</Aside>}}
+
+## Creating a Worker
+
+To use a Worker Variable, you first need to create a new Cloudflare Worker. You can do this through the Cloudflare dashboard or by using [Wrangler](https://developers.cloudflare.com/workers/get-started/guide/).
```
Comment (kodster28) [inline content/zaraz/advanced/worker-variables.md]:
```suggestion
To use a Worker Variable, you first need to create a new Cloudflare Worker. You can do this through the Cloudflare dashboard or by using [Wrangler](/workers/get-started/guide/).
```

File: content/zaraz/advanced/worker-variables.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+To create a new worker in the Cloudflare dashboard:
+
+1. Log in to the [Cloudflare dashboard](https://dash.cloudflare.com/).
+2. Go to **Workers**, and click on **Create a Service**.
```
Comment (kodster28) [inline content/zaraz/advanced/worker-variables.md]:
```suggestion
2. Go to **Workers** and select **Create a Service**.
```

File: content/zaraz/advanced/worker-variables.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+2. Give a name to your service and choose the **HTTP Handler** as your starter template.
+3. Click **Create Service**, and then **Quick Edit**.
+
+You have now created a basic Worker that responds with "Hello world." If you use this Worker as a Variable, your Variable will always output "Hello world." To make this Worker useful, you will need to use information coming from Zaraz, which is known as the Zaraz Context.
+
+Zaraz forwards the Zaraz Context object to your Worker as a JSON payload with a POST request. You can access any property like this:
+
+```
```
Comment (kodster28) [inline content/zaraz/advanced/worker-variables.md]:
```suggestion
```js
```

File: content/zaraz/advanced/worker-variables.md [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+/* System parameters */
+system.page.url.href // URL of the current page
+system.page.query.gclid // Value of the gclid query parameter
+system.device.resolution // Device screen resolution
+system.device.language // Browser preferred language
+
+/* Zaraz Track values */
+client.value // value from `zaraz.track("foo", {value: "bar"})`
+client.products[0].name // name of the first product in an ecommerce call
+
```
Comment (kodster28) [inline content/zaraz/advanced/worker-variables.md]:
```suggestion
```

File: content/zaraz/advanced/worker-variables.md [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+3. Select an action or add a new one.
+4. Click on the plus sign at the right of the text fields.
+5. Select your Worker Variable from the list.
+
+## Example Worker Variables
+
+
+### Calculates the sum of all products in the cart
+
+Assuming we're sending a list of products in a cart, like this:
```
Comment (kodster28) [inline content/zaraz/advanced/worker-variables.md]:
```suggestion
Assuming we are sending a list of products in a cart, like this:
```

File: content/zaraz/advanced/worker-variables.md [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+Assuming you're sending a value that your want to hash, for example, an email address:
+
+```js
+zaraz.track("user_logged_in", { email: "user@example.com" });
+```
+
+You can access this property and encrypt it like this:
+
+```js
+
```
Comment (kodster28) [inline content/zaraz/advanced/worker-variables.md]:
```suggestion
```

File: content/zaraz/advanced/worker-variables.md [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+  async fetch(request, env) {
+    // Parse the Zaraz Context object
+    const { system, client } = request.json();
+
+    const { email } = client;
+
+    return new Response(await digestMessage(text));
+  },
+};
+
```
Comment (kodster28) [inline content/zaraz/advanced/worker-variables.md]:
```suggestion
```

File: content/zaraz/faq.md [hunk 8]
```diff
@@ -17,7 +17,24 @@ Below you will find answers to our most commonly asked questions. If you cannot
 
 ### Setting up Zaraz
 
-#### Zaraz does not appear to be working. Is there a way to check where the problem is?
+#### Zaraz does not appear to be working.
+
+If you're experiencing issues with Zaraz, there could be multiple reasons behind it. First, it's important to verify that the Zaraz script is loading properly on your website.
```
Comment (kodster28) [inline content/zaraz/faq.md]:
```suggestion
If you are experiencing issues with Zaraz, there could be multiple reasons behind it. First, it's important to verify that the Zaraz script is loading properly on your website.
```

File: content/zaraz/reference/context.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+## System properties
+
+{{<table-wrap>}}
+
+| Property | Type | Description |
+| --- | --- | --- |
+| `system.page.query` | Object | Key-Value object containing all query parameters in the current URL. |
+| `system.page.title` | String | Current page title. |
+| `system.page.url` | URL | [URL](https://developer.mozilla.org/en-US/docs/Web/API/URL) Object containing information about the current URL |
+ `system.page.referrer` | String | Current page referrer from `document.referrer`. |
```
Comment (kodster28) [inline content/zaraz/reference/context.md]:
```suggestion
| `system.page.referrer` | String | Current page referrer from `document.referrer`. |
```

**Review by kodster28 — APPROVED** [review]

**Comment by bjesus** [comment]
Thank you @kodster28 for all the fixes!

File: content/zaraz/advanced/worker-variables.md [hunk 10]
```diff
@@ -0,0 +1,164 @@
...
+### Hash a value before sending it to a third-party vendor
+
+Assuming you're sending a value that your want to hash, for example, an email address:
+
+```js
+zaraz.track("user_logged_in", { email: "user@example.com" });
+```
+
+You can access this property and encrypt it like this:
```
Comment (WalshyDev) [inline content/zaraz/advanced/worker-variables.md]:
```suggestion
You can access this property and hash it like this:
```

File: content/zaraz/advanced/worker-variables.md [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+  const hashHex = hashArray
+    .map((b) => b.toString(16).padStart(2, "0"))
+    .join(""); // convert bytes to hex string
+  return hashHex;
+}
+
+export default {
+  async fetch(request, env) {
+    // Parse the Zaraz Context object
+    const { system, client } = request.json();
```
Comment (WalshyDev) [inline content/zaraz/advanced/worker-variables.md]: Are we getting the types added to `workers-types`? Would be great to have a TS example here with them

File: content/zaraz/advanced/worker-variables.md [hunk 12]
```diff
@@ -0,0 +1,164 @@
...
+2. Go to **Workers** and select **Create a Service**.
+2. Give a name to your service and choose the **HTTP Handler** as your starter template.
+3. Click **Create Service**, and then **Quick Edit**.
+
+You have now created a basic Worker that responds with "Hello world." If you use this Worker as a Variable, your Variable will always output "Hello world." To make this Worker useful, you will need to use information coming from Zaraz, which is known as the Zaraz Context.
+
+Zaraz forwards the Zaraz Context object to your Worker as a JSON payload with a POST request. You can access any property like this:
+
+```js
+const { system, client } = request.json()
```
Comment (WalshyDev) [inline content/zaraz/advanced/worker-variables.md]:
```suggestion
const { system, client } = await request.json()
```

File: content/zaraz/advanced/worker-variables.md [hunk 13]
```diff
@@ -0,0 +1,164 @@
...
+});
+```
+
+Calculating the sum can be done like this:
+
+```js
+export default {
+  async fetch(request, env) {
+    // Parse the Zaraz Context object
+    const { system, client } = request.json();
```
Comment (WalshyDev) [inline content/zaraz/advanced/worker-variables.md]:
```suggestion
    const { system, client } = await request.json();
```

File: content/zaraz/advanced/worker-variables.md [hunk 14]
```diff
@@ -0,0 +1,164 @@
...
+
+### Match a cookie with a user in your backend
+
+Zaraz exposes all cookies automatically under the `system.cookies` object, so they are always available. Accesing the cookie and using it to query your backend could look like this:
+
+```js
+export default {
+  async fetch(request, env) {
+    // Parse the Zaraz Context object
+    const { system, client } = request.json();
```
Comment (WalshyDev) [inline content/zaraz/advanced/worker-variables.md]:
```suggestion
    const { system, client } = await request.json();
```

File: content/zaraz/advanced/worker-variables.md [hunk 15]
```diff
@@ -0,0 +1,164 @@
...
+
+    // Get an array of all prices
```
Comment (WalshyDev) [inline content/zaraz/advanced/worker-variables.md]:
```suggestion
    // Get the value of the cookie "login-cookie"
```

**Comment by isaac-mcfadyen** [comment]
Might want ot mention something about the variable value being the Response from the Worker: I didn't understand that at first (assumed it was a binding) and had to dig through the examples for a bit.

File: content/zaraz/advanced/worker-variables.md (hunk 11 above)
Comment (bjesus) [reply content/zaraz/advanced/worker-variables.md]: great idea, we'll do that!

**Comment by bjesus** [comment]
Thank you @WalshyDev for all the fixes! and thanks @isaac-mcfadyen for pointing it out - I've just added a sentence that hopefully makes it clear.

**Review by WalshyDev — APPROVED** [review]

**Comment by kodster28** [comment]
Ready to merge in @bjesus? (not sure if you had permissions to do this yourself)

**Comment by bjesus** [comment]
yes, ready! I do not have permission to merge myself...

**Pull request merged**